src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -60,9 +60,11 @@ public void EndStreamValueScope()
         /// <returns>A task representing the asynchronous operation. The task result contains a stream for writing the stream value.</returns>
         public async Task<Stream> StartStreamValueScopeAsync()
         {
-            this.WriteSeparatorIfNecessary();
+            this.CommitUtf8JsonWriterContentsToBuffer();
+            this.WriteItemWithSeparatorIfNeeded();
             this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
-            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+
+            await this.FlushAsync().ConfigureAwait(false);

```

Comment:
Not a request for change, but I wonder why opted for `FlushAsync()` instead of `FlushIfBufferThresholdReachedAsync()` here.

Reply:
No reason really but I think it'll be good to flush when buffer threshold is reached instead of flushing every other time